### PR TITLE
Detaild treasury report

### PR DESCRIPTION
**Pull request by valdisz**
Game option `FACTION_STATISTICS` which extends `Faction Status` report section with detailed treasury report in the following way:

```
Faction Status:
Tax Regions: 0 (10)
Trade Regions: 0 (10)
Mages: 1 (1)
Treasury:                                 Rank  Max        Total
---------------------------------------------------------------------
101 leaders [LEAD]                        1     101        101
100 swords [SWOR]                         1     100        100
100 plate armor [PARM]                    1     100        100
```

### CODE REVIEW

File: basic/rules.cpp [hunk 1]
```diff
@@ -236,6 +236,7 @@ static GameDefs g = {
 	0, // STRICT_GUARD
 	1, // OCEAN_GUARD
 	BattleLogLevel::NORMAL,	// BATTLE_LOG_LEVEL
+	false,	//FACTION_STATISTICS
```
Comment (artyomtrityak) [inline basic/rules.cpp]: lets do same 1/0

File: game.cpp [hunk 2]
```diff
@@ -1174,6 +1174,18 @@ void Game::WriteReport()
 
 	MakeFactionReportLists();
 	CountAllSpecialists();
+
+    int ** citems = new int* [factionseq];
```
Comment (artyomtrityak) [inline game.cpp]: we should guard this code with feature flag as it is not need to be executed without Statistics enabled + indentation seems not right.

Any chance we can put _all_ events related code in a single file we can guard it execution by feature flag like we did for world events?

File: game.cpp [hunk 3]
```diff
@@ -2089,3 +2101,40 @@ void Game::WriteTimesArticle(AString article)
...
+        citems[i][j] = CountItem (fac, j);
+      }
+      i++;
+    }
+  }
+}
+
+int Game::CountItem (Faction * fac, int item)
+{
+  int all = 0;
```
Comment (artyomtrityak) [inline game.cpp]: those indentation does not seem correct

File: game.cpp [hunk 4]
```diff
@@ -2089,3 +2101,40 @@ void Game::WriteTimesArticle(AString article)
...
+      forlist (&(fac->present_regions))
+      {
+        ARegionPtr * r = (ARegionPtr *) elem;
+        forlist (&r->ptr->objects)
+        {
+          Object * obj = (Object *) elem;
```
Comment (artyomtrityak) [inline game.cpp]: i think it is 1-2 spaces instead of tab is used in other places, lets indent in a same way

File: game.h [hunk 5]
```diff
@@ -590,6 +590,10 @@ class Game
 	int CanAttack(ARegion *, AList *, Unit *);
 	void GetAFacs(ARegion *, Unit *, Unit *, AList &, AList &, AList &);
 	void GetDFacs(ARegion *, Unit *, AList &);
+
+	// For faction statistics
+    void CountItems (int **);
```
Comment (artyomtrityak) [inline game.h]: same for indentation

File: game.cpp (hunk 2 above)
Comment (valdisz) [reply game.cpp]: I will check, but this was ported from MagicDeep

File: game.cpp (hunk 3 above)
Comment (valdisz) [reply game.cpp]: That was working correctly for several years :)

File: game.cpp [hunk 6]
```diff
@@ -2091,19 +2106,57 @@ void Game::Equilibrate()
...
+	int all = 0;
+	forlist (&(fac->present_regions))
+	{
+		ARegionPtr * r = (ARegionPtr *) elem;
+		forlist (&r->ptr->objects)
+		{
+			Object * obj = (Object *) elem;
+			forlist (&obj->units)
+			{
+			Unit * unit = (Unit *) elem;
```
Comment (artyomtrityak) [inline game.cpp:2154]: this is not indented :)

**Review by artyomtrityak — APPROVED** [review]

**Pull request merged**